RecoLocalTracker/SiPixelRecHits/src/PixelCPEClusterRepair.cc

```diff
@@ -91,6 +94,55 @@ PixelCPEClusterRepair::PixelCPEClusterRepair(edm::ParameterSet const & conf,
    maxSizeMismatchInY_ = conf.getParameter<double>("MaxSizeMismatchInY");
    minChargeRatio_ = conf.getParameter<double>("MinChargeRatio");
 
+   // read sub-detectors to recommend 2D
+   // can be: 
+   //     XYX (XYZ = PXB, PXE)
+   //     XYZ n (XYZ as above, n = layer, wheel or disk = 1 .. 6 ;)
+   std::vector<std::string> str_Recommend2D = conf.getParameter<std::vector<std::string> >("Recommend2D");
+   Recommend2D_.reserve(str_Recommend2D.size());
+   for (std::vector<std::string>::const_iterator it = str_Recommend2D.begin(), ed = str_Recommend2D.end(); it != ed; ++it) {
+     Recommend2D_.push_back(Rule(*it));
+   }
+
+   // run CR on damaged clusters (and not only on edge hits)
+   RunDamagedClusters_ = conf.existsAs<bool>("RunDamagedClusters")?conf.getParameter<bool>("RunDamagedClusters"):false;
```

Comment:
Should you have defined a fillDescriptions() method in this class. this check wouldn't have been needed/
Defining such a method here shouldn't be complicated at all: however, if you don't want to do it now please keep it in mind for the post UL improvements/updates